native/swift/Sources/wordpress-api/SafeRequestExecutor.swift

```diff
@@ -18,83 +18,42 @@ extension SafeRequestExecutor {
 
 final class WpRequestExecutor: SafeRequestExecutor {
     private let session: URLSession
-    private let redirectTracker = RedirectTracker()
+    private let executorDelegate: RequestExecutorDelegate
 
-    init(urlSession: URLSession) {
+    private let additionalHttpHeadersForAllRequests: [String: String]
+
+    init(
+        urlSession: URLSession,
+        additionalHttpHeadersForAllRequests: [String: String] = [:]
```

Comment:
This argument is not used.

Reply:
Not by our code, but this is a good back-door way to add a user agent (for instance).

Should that just be a separate PR?

Reply:
The same feature can be achieved using `URLSession`. But I guess this argument would allow adding headers _only_ for API requests originated from the library. I'm okay with it being in this PR if it's needed.

Reply:
I'm hoping we can limit the number of things that a developer has to handle themselves in `URLSession`, but I guess we'll see?